JRuby --on-conflict=skip errors due to s3 not finding an output directory

Running jruby build with dry-run and on_conflict=skip:

```
gh workflow run build_jruby.yml     --repo heroku/docker-heroku-ruby-builder     --ref main     -f jruby_version=9.4.0.0     -f on_conflict=skip    -f dry_run=true
```

https://github.com/heroku/docker-heroku-ruby-builder/actions/runs/24792269134/job/72552645247

Produces an error because the output directory is not createdbut the s3 sync errors if it does not exist:

```
Run aws s3 sync ./output "s3://${S3_BUCKET}" --dryrun

Error: RROR]: The user-provided path ./output does not exist.
Error: Process completed with exit code 255.
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -196,6 +196,7 @@ jobs:
           grep -q "Already exists" "$LOGFILE"
           grep -q "skipping" "$LOGFILE"
           grep -q "status=skipped" "$METAFILE"
+          test -d ./output
       - name: "on-conflict=skip: skips when local file exists"
         run: |
           set -euo pipefail
@@ -211,6 +212,7 @@ jobs:
           grep -q "Output already exists locally" "$LOGFILE"
           grep -q "skipping" "$LOGFILE"
           grep -q "status=skipped" "$METAFILE"
+          test -d ./output
 
   check_inventory_urls:
     runs-on: ubuntu-24.04

--- a/jruby_executable/src/bin/jruby_build.rs
+++ b/jruby_executable/src/bin/jruby_build.rs
@@ -57,6 +57,7 @@ fn jruby_build(args: &Args) -> Result<BuildStatus, Box<dyn Error>> {
     let volume_output_dir = source_dir().join("output");
 
     fs::create_dir_all(&volume_cache_dir)?;
+    fs::create_dir_all(&volume_output_dir)?;
 
     let ruby_stdlib_version = jruby_build_properties(version)?.ruby_stdlib_version()?;
     let tgz_name = format!("ruby-{ruby_stdlib_version}-jruby-{version}.tgz");

